### PR TITLE
feature: counter increase function

### PR DESCRIPTION
**Pull request by ShayElkana**
https://monday.monday.com/boards/3670992828/views/83352826/pulses/5635984488

### CODE REVIEW

File: lib/storage/storage.ts [hunk 1]
```diff
@@ -1,9 +1,28 @@
...
+  Options,
+  Period,
+  SetResponse,
+  counterOptions
+} from 'types/storage';
+
 
 export class Storage extends BaseStorage implements IStorageInstance {
 
+  async increaseCounter(period: Period, options?: counterOptions) {
```
Comment (gregra81) [inline lib/storage/storage.ts:16]: incrementCounter

**Review by gregra81 — COMMENTED** [review]
Nice, left you 2 comments

**Review by gregra81 — APPROVED** [review]
💯

File: README.md [hunk 2]
```diff
@@ -68,6 +68,43 @@ const { success, error } = await storage.delete(key, { shared });
 
 </details>
 
+
+## Operations counter
+
+<details>
+<summary>A tool that provide the ability to count the number of operations base on a pre-defined period.</summary>
```
Comment (gregra81) [inline README.md:75]:
```suggestion
<summary>A tool that provides the ability to count the number of operations base on a pre-defined period.</summary>
```

File: README.md [hunk 3]
```diff
@@ -68,6 +68,43 @@ const { success, error } = await storage.delete(key, { shared });
...
+
+- This is the way to count operations around your project.
+- Has 3 pre-defined periods
+  - `DAILY` - the counter will be reset every day (base on GMT time zone)
+  - `MONTHLY` - the counter will be reset every month (base on GMT time zone).
+  - `YEARLY` -  the counter will be reset every year (base on GMT time zone).
```
Comment (gregra81) [inline README.md:81]: What does it mean `base on GMT time zone`?

File: README.md [hunk 4]
```diff
@@ -68,6 +68,43 @@ const { success, error } = await storage.delete(key, { shared });
...
+- You can increase the counter by a positive number e.g. if the current counter value is 4, and you like a specific operation to increase to 6 than you add 2 instead of the default which is 1, you can do that by using the BODY `incrementBy` property.
+- You can generate on the fly new counters e.g. for DOC caning and a different counter for IMAGE scanning by using the BODY `kind` property.
```
Comment (gregra81) [inline README.md:83]:
```suggestion
- You can increase the counter any positive number. By default it increases by 1
- We support different counter kinds, with a default counter that does not have any kind and is meant to simply count the usage of your workflow / app. If you want to count multiple types of operations e.g. for DOC scanning and  for IMAGE scanning you can supply the optional `kind` property.
```

File: README.md [hunk 5]
```diff
@@ -68,6 +68,43 @@ const { success, error } = await storage.delete(key, { shared });
...
+
+```typescript
+import { Storage } from '@mondaycom/apps-sdk';
+const storage = new Storage('<ACCESS_TOKEN>');
+```
+
+#### incrementCounter
+
+- `Period: string` - This is an enum for those value: `DAILY`, `MONTHLY`, `YEARLY`
+- `incrementBy?: number` - by how much to increase the current counter value (_OPTIONAL_)
```
Comment (gregra81) [inline README.md:95]:
```suggestion
- `incrementBy?: number` - by how much to increase the current counter value (_OPTIONAL_). Default is 1
```

File: README.md [hunk 6]
```diff
@@ -68,6 +68,43 @@ const { success, error } = await storage.delete(key, { shared });
...
+- `kind?: string` - set a new counter for the `kind` operation name (_OPTIONAL_)
+
+```typescript
+const { message, newCounterValue, success, error } = await storage.incrementCounter(Period.MONTHLY, {
+  incrementBy: 2,
+  kind: 'foo'
```
Comment (gregra81) [inline README.md:101]:
```suggestion
  kind: 'image.scan'
```

File: README.md (hunk 3 above)
Comment (ShayElkana) [reply README.md:81]: @gregra81 
Greenwich Mean Time (GMT) 
universal time
GMT 0
https://www.google.com/search?q=gmt+time+now&oq=hgmt&gs_lcrp=EgZjaHJvbWUqDwgEEAAYChiLAxixAxiABDIGCAAQRRg5MhIIARAAGAoYgwEYsQMYyQMYgAQyDwgCEAAYChiDARixAxiABDIMCAMQABgKGIsDGIAEMg8IBBAAGAoYiwMYsQMYgAQyEggFEAAYChiDARiLAxixAxiABDIJCAYQABgKGIAEMgkIBxAAGAoYgAQyCQgIEAAYChiABDIMCAkQABgKGJIDGIAE0gEINTIzNGoxajGoAgCwAgA&sourceid=chrome&ie=UTF-8

https://www.google.com/search?q=universal+time+now+VS+GMT&sca_esv=589592780&ei=JfR1ZfeQHp-lkdUPzcOcqAs&ved=0ahUKEwi3t4SlsYWDAxWfUqQEHc0hB7UQ4dUDCBA&uact=5&oq=universal+time+now+VS+GMT&gs_lp=Egxnd3Mtd2l6LXNlcnAiGXVuaXZlcnNhbCB0aW1lIG5vdyBWUyBHTVQyBRAhGKABMgUQIRigATIFECEYoAEyBRAhGKABMggQIRgWGB4YHUjpQ1CwFlibL3AHeAGQAQGYAc4DoAGAEaoBCTAuNC4yLjIuMbgBA8gBAPgBAcICChAAGEcY1gQYsAPCAg0QABiABBiKBRhDGLADwgIFEAAYgATCAgYQABgWGB7CAgoQIRgWGB4YDxgdwgIEECEYFeIDBBgAIEGIBgGQBgo&sclient=gws-wiz-serp

File: README.md (hunk 6 above)
Comment (ShayElkana) [reply README.md:101]: I limited the special chars to -_
"." is not included, So I set it to 

kind: 'image_scan'

File: README.md (hunk 3 above)
Comment (gregra81) [reply README.md:81]:
```
  - `DAILY` - the counter will be reset every day (at 00:00 GMT)
  - `MONTHLY` - the counter will be reset every month
  - `YEARLY` -  the counter will be reset every year
```

**Pull request merged**